exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/EndToEndTest.java

```diff
@@ -0,0 +1,151 @@
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.eq;
+
+@RunWith(JUnit4.class)
+public class EndToEndTest {
+  @Before
+  public void setup() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  // Creating a mock TraceServiceClient so that batchWriteSpans() method won't actually be called
+  TraceServiceClient mockTraceServiceClient = Mockito.mock(TraceServiceClient.class);
+
+  private TraceExporter exporter;
+
+  private static final String PROJECT_ID = "project-id";
+  private static final Map<String, AttributeValue> FIXED_ATTRIBUTES = new HashMap<>();
+  private static final TraceId TRACE_ID = new TraceId(321, 123);
+  private static final SpanId SPAN_ID = new SpanId(12345);
+  private static final SpanId PARENT_SPAN_ID = new SpanId(54321);
+  private static final TraceFlags TRACE_FLAGS = TraceFlags.builder().build();
+  private static final TraceState TRACE_STATE = TraceState.builder().build();
+  private static final SpanContext spanContext = SpanContext.create(TRACE_ID, SPAN_ID, TRACE_FLAGS, TRACE_STATE);
+  private static final String SPAN_NAME = "MySpanName";
+  private static final long START_EPOCH_NANOS = TimeUnit.SECONDS.toNanos(3000) + 200;
+  private static final long START_SECONDS = TimeUnit.NANOSECONDS.toSeconds(START_EPOCH_NANOS);
+  private static final int START_NANOS = (int) (START_EPOCH_NANOS - TimeUnit.SECONDS.toNanos(START_SECONDS));
+  private static final long END_EPOCH_NANOS = TimeUnit.SECONDS.toNanos(3001) + 255;
+  private static final long END_SECONDS = TimeUnit.NANOSECONDS.toSeconds(END_EPOCH_NANOS);
+  private static final int END_NANOS = (int) (END_EPOCH_NANOS - TimeUnit.SECONDS.toNanos(END_SECONDS));
+  private static final String SERVER_PREFIX = "Recv.";
+  private static final String AGENT_LABEL_KEY = "g.co/agent";
+  private static final String OPEN_TELEMETRY_LIBRARY_VERSION = "0.3.0";
+  private static final String AGENT_LABEL_VALUE_STRING = "opentelemetry-java [" + OPEN_TELEMETRY_LIBRARY_VERSION + "]";
+  private static final AttributeValue AGENT_LABEL_VALUE = AttributeValue.newBuilder()
+          .setStringValue(TruncatableString.newBuilder().setValue(AGENT_LABEL_VALUE_STRING).setTruncatedByteCount(0).build()).build();
+  private static final Span.Attributes ATTRIBUTES = Span.Attributes.newBuilder()
+          .setDroppedAttributesCount(0)
+          .putAttributeMap(AGENT_LABEL_KEY, AGENT_LABEL_VALUE)
+          .build();
+  private static final Span.TimeEvents TIME_EVENTS = Span.TimeEvents.newBuilder().build();
+  private static final io.opentelemetry.trace.Status SPAN_DATA_STATUS = io.opentelemetry.trace.Status.OK;
+  private static final Status SPAN_STATUS = Status.newBuilder().setCode(SPAN_DATA_STATUS.getCanonicalCode().value())
+          .build();
+  private static final Span.Links LINKS = Span.Links.newBuilder().setDroppedLinksCount(0).build();
+
+
+  // Some test cases. Since the core function of this exporter is export(), we may focus on testing that.
```

Comment:
Some comments from the example code were more likely guidances on how to write unit test, consider remove it in production code. 

Here's a great quote from Robert C. Martin's Clean Code:

_The proper use of comments is to compensate for our failure to express ourself in code. Note that I used the word failure. I meant it. Comments are always failures. We must have them because we cannot always figure out how to express ourselves without them, but their use is not a cause for celebration.... Every time you express yourself in code, you should pat yourself on the back. Every time you write a comment, you should grimace and feel the failure of your ability of expression._

Reply:
Thanks for the advice! I don't have too much exposure to production code so I'm glad to find out things like this. That's a really good quote too, will be sure to not include unnecessary comments in my code in the future